### PR TITLE
CI/Docs: Add jobs to deploy component docs

### PR DESCRIPTION
**Pull request by david-cermak**
Added the build-docs job to:
* generate `breathe` style documentation
* deploy it to `gh-pages` branch

To be published as GitHub pages.

### CODE REVIEW

**Comment by david-cermak** [comment]
@kumekay PTAL, example of the generated docs is available [here](https://david-cermak.github.io/esp-protocols/esp_modem/index.html)

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          sudo apt-get update
+          sudo apt-get -y install doxygen clang python3-pip
+          python -m pip install breathe recommonmark
+          cd $GITHUB_WORKSPACE/components/esp_modem/docs
+          ./generate_docs
+          cd $GITHUB_WORKSPACE/docs
+          touch .nojekyll
+          echo '<a href="esp_modem/index.html">esp-modem</a>' > index.html
+
+      - name: Commit generated docs
```
Comment (kumekay) [inline .github/workflows/build-docs.yml:31]: There is a separate action for publishing on gh-pages in one step:
https://github.com/JamesIves/github-pages-deploy-action

It should make the job a bit cleaner.

**Comment by kumekay** [comment]
Docs look good to me, a short not on GitHub pages deployment

File: .github/workflows/build-docs.yml (hunk 1 above)
Comment (david-cermak) [reply .github/workflows/build-docs.yml]: Thanks, used the two in one action!

**Comment by david-cermak** [comment]
Thanks for the suggestions, Sergei.

Also added a docs build step before the component deployment to the registry, so the documentation is generated also locally when component is downloaded.

**Review by kumekay — APPROVED** [review]
Thank you for update, LGTM

File: .github/workflows/upload_component.yml [hunk 2]
```diff
@@ -12,10 +9,10 @@ jobs:
       - uses: actions/checkout@master
 
       - name: Upload components to component service
-        uses: espressif/github-actions/upload_components@master
+        uses: david-cermak/github-actions/upload_components@feature/upload_component_hooks
```
Comment (kumekay) [inline .github/workflows/upload_component.yml:12]: Just don't forget to revert this, before merging

Comment (david-cermak) [reply .github/workflows/upload_component.yml:12]: :-) Thanks for reminder, I will. Forgot to mark it *Draft* while some WIP commits still present.

Comment (david-cermak) [reply .github/workflows/upload_component.yml]: Done.

Did it as @igrr suggested in the referenced PR. This way it's less code and runs faster (builds the docs just once and uses it for both the `gh-pages` and the component registry), so I think it's okay as we have just one component.

**Review by kumekay — APPROVED** [review]

**Comment by david-cermak** [comment]
Thanks for the review!

**Pull request merged**